resources/web/__test__/air_gapped_template.test.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const { promisify } = require('util');
+
+const readdir = promisify(fs.readdir);
+const readFile = promisify(fs.readFile);
+
+describe("air_gapped_template.html", () => {
+  let template;
+  beforeAll(async () => {
+    template = await readFile("air_gapped_template.html", {encoding: "utf8"});
+  });
+  describe("links", () => {
+    /**
+     * Outgoing links from the template.
+     */
+    let links;
+    beforeAll(() => {
+      links = Array.from(new Set([...template.matchAll(/(?:href|src)="([^"]+)"/g)].map(m => m[1]))).sort();
```

Comment:
What's the value of `...` spreading into a fresh array? Does `template.matchAll` return a pseudo-array?

Reply:
An iterator. Let me check if that works actually. I had the spread before the set.

Reply:
I played with it a bit and the `[...]` is required because you can't call `map` on the iterator that `matchAll` returns.